src/app/Topology/Toolbar/TopologyFilters.tsx

```diff
@@ -272,7 +272,10 @@ export const TopologyFilter: React.FC<{ isDisabled?: boolean }> = ({ isDisabled,
             value={{
               toString: () => opt,
               compareTo: (other) => {
-                const regex = new RegExp(typeof other === 'string' ? other : other.value, 'i');
+                const regex = new RegExp(
+                  typeof other === 'string' ? other : isKeyValue(other) ? other.value : `${other}`,
+                  'i',
+                );
                 return regex.test(opt);
               },
               ...{
```

Comment:
It could be good to truncate the label in case it is too long (e.g. URL). On Patternfly 5, we can control the number of characters. 

```diff
diff --git a/src/app/Topology/Toolbar/TopologyFilters.tsx b/src/app/Topology/Toolbar/TopologyFilters.tsx
index ddb4109..f84ea3d 100644
--- a/src/app/Topology/Toolbar/TopologyFilters.tsx
+++ b/src/app/Topology/Toolbar/TopologyFilters.tsx
@@ -213,7 +213,7 @@ export const TopologyFilter: React.FC<{ isDisabled?: boolean }> = ({ isDisabled,
                   },
                 }}
               >
-                {isLabelOrAnnotation(cat) ? <Label color="grey">{opt}</Label> : opt}
+                {isLabelOrAnnotation(cat) ? <Label color="grey" isTruncated>{opt}</Label> : opt}
               </SelectOption>
             ))}
           </SelectGroup>
@@ -286,7 +286,7 @@ export const TopologyFilter: React.FC<{ isDisabled?: boolean }> = ({ isDisabled,
               },
             }}
           >
-            {isLabelOrAnnotation(cat) ? <Label color="grey">{opt}</Label> : opt}
+            {isLabelOrAnnotation(cat) ? <Label color="grey" isTruncated>{opt}</Label> : opt}
           </SelectOption>
         );
       });
```

Reply:
Without truncating, it looks like:

![image](https://github.com/user-attachments/assets/9f9fc620-fb73-4ff3-bce4-b0142ae68c7f)

Reply:
With truncation it looks really short:

![image](https://github.com/user-attachments/assets/51afcb93-f117-4fb3-94aa-4ab5ef30acc3)

I'll try some other approach to truncation, this pf5 component attribute doesn't seem quite right. At least it does give the full text on hover.

Reply:
There is a `textMaxWidth` attribute documented in 5.3.0, so maybe this can be fixed up in the pf5 branch later.

Reply:
I'll leave that change out of this PR since I don't want that to get backported without the max width control.

Reply:
Right, I used that `textMaxWidth` in #1308 ^^ Seems like we acn specify things like `16ch`.